bls/src/test/java/tech/pegasys/teku/bls/BLSSignatureTest.java

```diff
@@ -84,14 +85,34 @@ void succeedsWhenEqualsReturnsFalseForDifferentSignatures() {
   @Test
   void succeedsWhenRoundtripSSZReturnsTheSameSignature() {
     BLSSignature signature1 = BLSSignature.random(65);
-    BLSSignature signature2 = BLSSignature.fromBytes(signature1.toBytes());
+    BLSSignature signature2 = BLSSignature.fromSSZBytes(signature1.toSSZBytes());
     assertEquals(signature1, signature2);
   }
 
   @Test
   void succeedsWhenRoundtripSSZReturnsTheEmptySignature() {
     BLSSignature signature1 = BLSSignature.empty();
-    BLSSignature signature2 = BLSSignature.fromBytes(signature1.toBytes());
+    BLSSignature signature2 = BLSSignature.fromSSZBytes(signature1.toSSZBytes());
     assertEquals(signature1, signature2);
   }
+
+  @Test
+  void succeedsWhenEqualsReturnsTrueForEmptySignatures() {
+    assertEquals(BLSSignature.empty(), BLSSignature.empty());
+    assertEquals(BLSSignature.empty().hashCode(), BLSSignature.empty().hashCode());
+  }
+
+  @Test
+  void roundtripEncodeDecodeCompressed() {
+    BLSSignature signature = BLSSignature.random(513);
+    final BLSSignature result = BLSSignature.fromBytesCompressed(signature.toBytesCompressed());
+    assertEquals(signature, result);
+    assertEquals(signature.hashCode(), result.hashCode());
+  }
+
+  @Test
+  void testRandomFromSeed() {
+    BLSSignature randomSig = BLSSignature.random(92892840);
+    System.out.println(randomSig);
```

Comment:
Should probably assert something here - ideally that the signature is a member of G2. Or that we can sign and verify something with it. Not sure really.